apps/condo/domains/billing/schema/RegisterBillingReceiptsService.js

```diff
@@ -0,0 +1,456 @@
+            receiptsToAdd.push(item)
+        } else {
+            const existingReceiptByKey = receiptsIndex[receiptKey]
+
+            const existingToPay = new Big(existingReceiptByKey.toPay)
+            const newToPay = new Big(item.toPay)
+            const toPayIsEqual = existingToPay.eq(newToPay)
+
+            const existingServices = existingReceiptByKey.services
+            // if not specified, then it is null
+            const newServices = item.services ? item.services : null
+            const servicesIsEqual = _.isEqual(existingServices, newServices)
+
+            const existingToPayDetails = existingReceiptByKey.toPayDetails
+            // if not specified, then it is null
+            const newToPayDetails = item.toPayDetails ? item.toPayDetails : null
+            const toPayDetailsIsEqual = _.isEqual(existingToPayDetails, newToPayDetails)
+
+            const shouldUpdateReceipt = !toPayIsEqual || !servicesIsEqual || !toPayDetailsIsEqual
+
+            if (shouldUpdateReceipt) {
+                item.id = existingReceiptByKey.id
+                receiptsToUpdate.push(item)
+            } else {
+                notChangedReceipts.push(item)
+            }
+        }
+    })
+
+    const newReceipts = []
+    for (const item of receiptsToAdd) {
+        const billingReceiptGQLInput = convertBillingReceiptToGQLInput(item, propertiesIndex, accountsIndex)
+        const newReceipt = await BillingReceipt.create(context, billingReceiptGQLInput)
+        newReceipts.push(newReceipt)
+    }
+
+    const updatedReceipts = []
+    for (const item of receiptsToUpdate) {
+        const itemId = item.id
+        const billingReceiptGQLInput = convertBillingReceiptToGQLInput(item, propertiesIndex, accountsIndex)
+        const updatableItem = _.omit(billingReceiptGQLInput, ['context', 'id'])
+        const updatedReceipt = await BillingReceipt.update(context, itemId, updatableItem)
+        updatedReceipts.push(updatedReceipt)
+    }
+
+    return { createdReceipts: newReceipts, updatedReceipts: updatedReceipts, notChangedReceipts: notChangedReceipts }
+}
+
+const RegisterBillingReceiptsService = new GQLCustomSchema('RegisterBillingReceiptsService', {
+    types: [
+        {
+            access: true,
+            type: 'input RegisterBillingReceiptInput ' +
+                '{ ' +
+                    'importId: String! ' +
+
+                    'address: String! ' +
+                    'normalizedAddress: String ' +
+
+                    'accountNumber: String! ' +
+                    'unitName: String! ' + // Is going to be made optional in future
+                    'unitType: String! ' + // Is going to be made optional in future
+                    'fullName: String ' +
+
+                    'toPay: String! ' +
+                    'toPayDetails: BillingReceiptServiceToPayDetailsFieldInput ' +
+                    'services: [BillingReceiptServiceFieldInput] ' +
+
+                    'month: Int! ' +
```

Comment:
? why int?
not just period

Reply:
We have an implicit rule, that period should be in format `year-month-01` 

I think this is more user-friendly to set month and year to int, then to support this bizarre rule

Reply:
Probably, it's better to use the same type as in billing rather than create a new rule. And you also need to decide should id start with `0` or `1`. It's better to use user friendly error about the format